src/server/game/Entities/Object/Object.cpp

```diff
@@ -815,7 +816,7 @@ void Object::BuildDynamicValuesUpdate(uint8 updateType, ByteBuffer* data, Player
             data->resize(data->size() + arrayBlockCount * sizeof(UpdateMask::BlockType));
             for (std::size_t v = 0; v < values.size(); ++v)
             {
-                if (updateType == UPDATETYPE_VALUES ? _dynamicChangesArrayMask[index][v] : values[v])
+                if (updateType != UPDATETYPE_VALUES || _dynamicChangesArrayMask[index][v])
```

Comment:
I understand, but in my sniffs there is a lot of DYNAMIC_FIELD send even if value is 0 at creation.

Conversation does not work if 0 value is not sent, so this is mandatory, but of course there is perhaps a better way to do it